src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java

```diff
@@ -443,4 +446,131 @@ public void doRemoveSidFromAgentRole(@QueryParameter(required = true) String rol
         FolderAuthorizationStrategyAPI.removeSidFromAgentRole(sid, roleName);
         redirect();
     }
+
+    /**
+     * API Method to get an {@link AgentRole}
+     * Example: {@code curl -X GET 'http://localhost:8080/jenkins/folder-auth/getAgentRole?name=agentSmithRole}
+     *
+     * @param name name of the role (single, no list)
+     */
+    @Restricted(NoExternalUse.class)
+    public void doGetAgentRole(@QueryParameter(required = true) String name) throws IOException {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        JSONObject responseJson = new JSONObject();
```

Comment:
I think we can pass a `true` to the JSONObject constructor so that we return null instead of an empty object (`{}`). Or maybe we can return a descriptive error message saying that role does not exist.

http://json-lib.sourceforge.net/apidocs/jdk15/net/sf/json/JSONObject.html#JSONObject()

Reply:
Hmm..I would prefer not to return an error message and instead return either an empty object or null as I find it more inline with other services. I can change it to return null